src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -46832,6 +46832,206 @@ class CL1
                 Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x1").WithLocation(11, 15));
         }
 
+
+        /// <remarks>Ported from <see cref="FlowTests.IfConditionalConstant" />.</remarks>
+        [Theory]
+        [InlineData("true", "false")]
+        [InlineData("FIELD_TRUE", "FIELD_FALSE")]
+        [InlineData("LOCAL_TRUE", "LOCAL_FALSE")]
+        [InlineData("true || false", "true && false")]
+        [InlineData("!false", "!true")]
+        public void IfConditionalConstant(string @true, string @false)
+        {
+            var source = @"
+#pragma warning disable 219 // The variable is assigned but its value is never used
+
+class C
+{
+    const bool FIELD_TRUE = true;
+    const bool FIELD_FALSE = false;
+
+    static void M(bool b, object? x, object y)
+    {
+        const bool LOCAL_TRUE = true;
+        const bool LOCAL_FALSE = false;
+
+        {
+            x = null;
+            if (b ? x != null : " + @false + @")
+                y = x;
+        }
+
+        {
+            x = null;
+            if (b ? x != null : " + @true + @")
+                y = x; // 1
+        }
+
+        {
+            x = null;
+            if (b ? " + @false + @" : x != null)
+                y = x;
+        }
+
+        {
+            x = null;
+            if (b ? " + @true + @" : x != null)
```

Comment:
Consider adding variations where the ternay is `!`'d.

Reply:
Does this mean wrapping the whole thing, i.e. `!(b ? true : x != null)`? Or something else?

Reply:
Exactly what you said.

Reply:
Done